dashboard/db/migrate/20210907200327_convert_application_classes.rb

```diff
@@ -0,0 +1,60 @@
+class ConvertApplicationClasses < ActiveRecord::Migration[5.2]
+  TEACHER_APPLICATION_CLASSES = {
+    "2018-2019" => "Pd::Application::Teacher1819Application",
+    "2019-2020" => "Pd::Application::Teacher1920Application",
+    "2020-2021" => "Pd::Application::Teacher2021Application",
+    "2021-2022" => "Pd::Application::Teacher2122Application"
+  }
+  PRINCIPAL_APPROVAL_APPLICATION_CLASSES = {
+    "2018-2019" => "Pd::Application::PrincipalApproval1819Application",
+    "2019-2020" => "Pd::Application::PrincipalApproval1920Application",
+    "2020-2021" => "Pd::Application::PrincipalApproval2021Application",
+    "2021-2022" => "Pd::Application::PrincipalApproval2122Application"
+  }
+
+  # Application subclasses were consolidated from year specific classes to a single application class
+  # Rails stores a db field 'type' which indicates which ruby class should be associated with the db entry,
+  # so these types need to be updated to match the classes that exist.
+  def change
+    reversible do |dir|
+      dir.up do
+        # convert Teacher1819Application to TeacherApplication
+        ActiveRecord::Base.transaction do
+          Pd::Application::ApplicationBase.
```

Comment:
Sorry for commenting late, but are we worried about this migration taking a long time? I figured the table isn't super large but I'm not sure

Reply:
there are 42k entries in pd_applications, so I'm not super concerned about this migration, especially the up direction which uses update_all which i think is more performant

Reply:
Yeah that seems fine to me! I did a migration with an `update_all` recently that ran into problems, but that table had a few million entries.

Reply:
migration took ~10 seconds on a production clone